File: 이것이 코딩테스트다/05.이진 탐색/3.반복문으로 구현한 이진탐색.js
const binarySearch = (array, target, start, end) => {
  while (start <= end) {
    mid = Math.floor((start + end) / 2);
    if (target === array[mid]) {
      return mid + 1;
    } else if (target < array[mid]) {
      end = mid - 1;
    } else {
      start = mid + 1;
    }
  }
  return NaN;
};

File: 이것이 코딩테스트다/05.이진 탐색/3.반복문으로 구현한 이진탐색.py
def binary_search(array, target, start, end):
  while start<=end:
    mid = (end + start) // 2

    if array[mid] < target:
      start = mid+1
    elif array[mid] > target:
      end = mid-1
    else:
      return mid+1
  return None

n, target = map(int, input().split())
array = list(map(int, input().split()))
print(binary_search(array, target, 0, n))